Sources/RateScreen/RateScreen.swift

```diff
@@ -0,0 +1,264 @@
+import SwiftUI
+
+struct RateScreen: View {
+    
+    private let columns = [GridItem(.flexible())]
+    @State private var selectedDateStart = Date()
+    @State private var selectedDateEnd = Date()
+    @State private var reload = false
+    
+    var chosenCurShortName: String
+    var chosenCurShortName1: String?
+    var chosenCurShortName2: String?
+    var chosenCurShortName3: String?
+    var apiInputTF: String
+    
+    @State private var conversion2Results: [String] = []
+    @State private var conversion3Results: [String] = []
+    @State private var conversion4Results: [String] = []
+    
+    let currencyApi = CurrencyApi()
+    
+    private let titles = ["Date", "#1", "#2", "#3"]
+    
+    private var converterScreen = ConverterScreen()
+    
+    private var dateRange: [String] {
+        let calendar = Calendar.current
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+        
+        var dateArray: [String] = []
+        var currentDate = calendar.startOfDay(for: selectedDateStart)
+        let endDate = calendar.startOfDay(for: selectedDateEnd)
+        
+        while currentDate <= endDate {
+            dateArray.append(dateFormatter.string(from: currentDate))
+            guard let nextDate = calendar.date(byAdding: .day, value: 1, to: currentDate) else { break }
+            currentDate = nextDate
+        }
+        
+        return dateArray
+    }
+    
+    init(chosenCurShortName: String, chosenCurShortName1: String?, chosenCurShortName2: String?, chosenCurShortName3: String?, apiInputTF: String) {
+        self.chosenCurShortName = chosenCurShortName
+        self.chosenCurShortName1 = chosenCurShortName1
+        self.chosenCurShortName2 = chosenCurShortName2
+        self.chosenCurShortName3 = chosenCurShortName3
+        self.apiInputTF = apiInputTF
+        
+        let appearance = UINavigationBarAppearance()
+        appearance.backgroundColor = .black
+        appearance.configureWithOpaqueBackground()
+        appearance.backgroundColor = .black
+        //appearance.configureWithTransparentBackground()
+        appearance.titleTextAttributes = [
+            .font: UIFont.systemFont(ofSize: 24),
+            .foregroundColor: UIColor.white
+        ]
+        UINavigationBar.appearance().standardAppearance = appearance
+        UINavigationBar.appearance().scrollEdgeAppearance = appearance
+    }
+    
+    var body: some View {
+        
+        NavigationView {
+            
+            ZStack {
```

Comment:
Why `ZStack` here?

Reply:
Because I put backgroundColor here and inside of that is NavigationView as an object.

Reply:
And without `ZStack` it does not work?